src/main/java/com/guysfromusa/carsgame/control/CommandConsumer.java

```diff
@@ -43,23 +47,16 @@ public synchronized void handle(@SuppressWarnings("unused") CommandEvent event)
 
             queuesNotEmpty = gameToPlayRoundOptional.isPresent();
 
-            gameToPlayRoundOptional.ifPresent(gameState -> {
-                gameState.setRoundInProgress(true);
-
-                //FIXME filter messages by type and group different cars
-                List<Command> consumedCommands = new ArrayList<>();
-                while (!gameState.getCommandsQueue().isEmpty()) {
-                    consumedCommands.add(gameState.getCommandsQueue().poll());
-                }
-
-                consumedCommands.stream()
-                        .collect(groupingBy(Command::getMessageType))
-                        .forEach((messageType, commands) ->
-                                handle(messageType, commands, gameState.getGameName()));
-            });
+            gameToPlayRoundOptional.ifPresent(gameState -> triggerRound(gameState));
         }
     }
 
+    private void triggerRound(GameState gameState) {
+        gameState.setRoundInProgress(true);
+        gameRoundSelector.selectCommand(gameState.getCommandsQueue(), gameState.getGameName())
+                .ifPresent(gameRound -> handle(gameRound.getMessageType(), gameRound.getCommands(), gameRound.getGameName()));
```

Comment:
setRoundInProgress should be inside ifPresent I think -> otherwise nobody will restore it to false

Reply:
actually this optional is redundant (now I see it)
because selected game have not empty queue